README.md

```diff
@@ -148,86 +168,166 @@ Klaviyo.createEvent(event)
-    super.onNewIntent(intent)
+    /* ... */
 
     // Tracks when a system tray notification is opened
     Klaviyo.handlePush(intent)
 }
 ```
 
-To specify a notification icon, add the following metadata to your app manifest.
-Absent this, the application's launcher icon will be used.
+**Note** intent handling may differ depending on your app's architecture. Adjust this example to your use-case, 
+ensuring that `Klaviyo.handlePush(intent)` is called when your app is opened from a notification.
+
+#### Deep Linking 
+**Deep Links** allow you to navigate to a particular page within your app in response to the user 
+opening a notification. There are broadly three steps to implement deep links in your app. 
+
+##### Add intent filters for incoming links
+Add an intent filter to the activity element of your `AndroidManifest.xml` file.
+Replace the scheme and host to match the URI scheme that you will embed in your push notifications. 
 
 ```xml
-<meta-data android:name="com.klaviyo.push.default_notification_icon"
-    android:resource="{YOUR_ICON_RESOURCE}" />
+<intent-filter android:label="@string/filter_view_example_gizmos">
+    <action android:name="android.intent.action.VIEW" />
+    <category android:name="android.intent.category.DEFAULT" />
+    <category android:name="android.intent.category.BROWSABLE" />
+    <!-- Accepts URIs formatted "example://host.com” -->
+    <data android:scheme="example" android:host="host.com"/>
+</intent-filter>
 ```
 
-### Manual implementation of `FirebaseMessagingService` (Advanced)
+##### Read data from incoming intents 
+When the app is opened from a deep link, the intent that started the activity contains data for the deep link.
+You can parse the URI from the intent's data property and use it to navigate to the appropriate part of your app. 
+
+```kotlin
+override fun onCreate(savedInstanceState: Bundle?) {
+    /* ... */
+    
+    onNewIntent(intent)
+}
 
+override fun onNewIntent(intent: Intent?) {
+    // Tracks when a system tray notification is opened
+    Klaviyo.handlePush(intent)
+   
+    // Read deep link data from intent
+    val action: String? = intent?.action 
+    val deepLink: Uri? = intent?.data
+}
+```
+
+##### Test your deep links
+Using [android debug bridge (adb)](https://developer.android.com/studio/command-line/adb), run the following command to launch your app via an intent
+containing a deep link to test your deep link handler.
+
+```shell
+adb shell am start
+    -W -a android.intent.action.VIEW
+    -d <URI> <PACKAGE>
+```
+
+To perform integration testing, you can send a
+[preview push notification](https://help.klaviyo.com/hc/en-us/articles/18011985278875) 
+containing a deep link from the Klaviyo push editor. 
+
+For additional resources on deep linking, refer to
+- [Android developer documentation](https://developer.android.com/training/app-links/deep-linking)
+- [Klaviyo help center](https://help.klaviyo.com/hc/en-us/articles/14750403974043)
+
+### Advanced Setup
```

Comment:
Is this intended to be a generic section for advanced, optional steps across the SDK - or just advanced steps for the push notifications section?
Just checking to ensure the header weight is at the correct level for the intention.

Reply:
Advanced push setup yeah